src/common/tracer.h

```diff
@@ -0,0 +1,67 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include <cstring>
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+#include <atomic>
+
+#include <jaegertracing/Tracer.h>
+#include <jaegertracing/net/IPAddress.h>
+#include <jaegertracing/net/Socket.h>
+
+
+namespace jaeger_tracing{
+
+  typedef opentracing::Span Span;
+
+  static std::shared_ptr<opentracing::v3::Tracer> tracer = NULL;
+
+  static inline void init_tracer(const char* tracerName,const char* filePath){
+        try{
+            auto yaml = YAML::LoadFile(filePath);
+            auto configuration = jaegertracing::Config::parse(yaml);
+            tracer = jaegertracing::Tracer::make(
+            tracerName,
+            configuration,
+            jaegertracing::logging::consoleLogger());
+        }catch(...) {
+          return;
```

Comment:
please add error handling. add error message to debug log

Reply:
noted